yt_dlp/extractor/kaltura.py

```diff
@@ -169,30 +171,36 @@ def _extract_urls(webpage):
 
     def _kaltura_api_call(self, video_id, actions, service_url=None, *args, **kwargs):
         params = actions[0]
-        if len(actions) > 1:
-            for i, a in enumerate(actions[1:], start=1):
-                for k, v in a.items():
-                    params['%d:%s' % (i, k)] = v
+        params.update({i: a for i, a in enumerate(actions[1:], start=1)})
 
         data = self._download_json(
             (service_url or self._SERVICE_URL) + self._SERVICE_BASE,
-            video_id, query=params, *args, **kwargs)
+            video_id, data=json.dumps(params).encode('utf-8'),
+            headers={
+                "Content-Type": "application/json",
+                "Accept-Encoding": "gzip, deflate, br",
+            }, *args, **kwargs)
 
-        status = data if len(actions) == 1 else data[0]
-        if status.get('objectType') == 'KalturaAPIException':
-            raise ExtractorError(
-                '%s said: %s' % (self.IE_NAME, status['message']))
+        for idx, status in enumerate(data):
+            if not isinstance(status, dict):
+                continue
+            if status.get('objectType') == 'KalturaAPIException':
+                raise ExtractorError(
+                    '%s said: %s (%d)' % (self.IE_NAME, status['message'], idx))
+
+        info_list = data[1]['objects']
+        data[1] = info_list and info_list[0]
```

Comment:
why not just
```suggestion
        data[1] = data[1]['objects'][0]
```
OR
```suggestion
        data[1] = traverse_obj(data, (1, 'objects', 0))
```

It seems weird to me that the access till `objects` is fatal, but then the `0` is non-fatal

Reply:
`data[1]['objects']` returns a list that might be empty. But the API call itself could be valid. That's why there is this check in  line 306, 307. Should it be handled here?